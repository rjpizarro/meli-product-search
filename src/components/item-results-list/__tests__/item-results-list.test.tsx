import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react';
import ItemResultsList from '../index'

const mockItemClick = jest.fn()

jest.mock('../../../components/item-card', () => () => <div data-testid="item-card" />)

const items = [
    {
        id: "MOCK-ID-1",
        title: "Test item 1",
        condition: "new",
        description: "Test description 1",
        picture: "test picture 1",
        sold_quantity: 10,
        price: {
            amount: 100,
            decimals: 0
        }
    },
    {
        id: "MOCK-ID-2",
        title: "Test item 2",
        condition: "new",
        description: "Test description 2",
        picture: "test picture 2",
        sold_quantity: 10,
        price: {
            amount: 100,
            decimals: 0
        }
    },
    {
        id: "MOCK-ID-3",
        title: "Test item 4",
        condition: "new",
        description: "Test description 3",
        picture: "test picture 3",
        sold_quantity: 10,
        price: {
            amount: 100,
            decimals: 0
        }
    },
    {
        id: "MOCK-ID-4",
        title: "Test item 4",
        condition: "new",
        description: "Test description 4",
        picture: "test picture 4",
        sold_quantity: 10,
        price: {
            amount: 100,
            decimals: 0
        }
    },
]

describe('ItemResultsList', () => {
    it('should render a list of item-card ', () => {
        const tree = renderer
            .create(<ItemResultsList onItemClick={mockItemClick} results={items} />)
            .toJSON()
        const { getAllByTestId }  = render(<ItemResultsList onItemClick={mockItemClick} results={items} />);

        const cardButtons = getAllByTestId('button')

        expect(tree).toMatchSnapshot()
        fireEvent.click(cardButtons[0])

        expect(mockItemClick).toHaveBeenCalledTimes(1)
        expect(mockItemClick).toHaveBeenCalledWith("MOCK-ID-1")
        expect(cardButtons).toHaveLength(4)
    })
})

