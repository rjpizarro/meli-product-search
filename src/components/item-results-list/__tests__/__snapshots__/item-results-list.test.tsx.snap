// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemResultsList should render a list of item-card  1`] = `
<ul
  className="item-results-list"
>
  <li
    className="item-results-list__card"
  >
    <button
      className="item-results-list__button"
      data-testid="button"
      onClick={[Function]}
    >
      <div
        data-testid="item-card"
      />
    </button>
  </li>
  <li
    className="item-results-list__card"
  >
    <button
      className="item-results-list__button"
      data-testid="button"
      onClick={[Function]}
    >
      <div
        data-testid="item-card"
      />
    </button>
  </li>
  <li
    className="item-results-list__card"
  >
    <button
      className="item-results-list__button"
      data-testid="button"
      onClick={[Function]}
    >
      <div
        data-testid="item-card"
      />
    </button>
  </li>
  <li
    className="item-results-list__card"
  >
    <button
      className="item-results-list__button"
      data-testid="button"
      onClick={[Function]}
    >
      <div
        data-testid="item-card"
      />
    </button>
  </li>
</ul>
`;
